core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -583,7 +585,17 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           _ => Future.successful(false)
         }
 
-      adjustTotalExecutors.flatMap(killExecutors)(ThreadUtils.sameThread)
+      val killResponse = adjustTotalExecutors.flatMap(killExecutors)(ThreadUtils.sameThread)
+
+      def execKilled(killStatus: Boolean): Future[Seq[String]] = Future.successful(
+        if (killStatus) {
+          executorsToKill
+        } else {
+          Seq.empty[String]
+        }
+      )
+
+      killResponse.flatMap(flag => execKilled(flag))(ThreadUtils.sameThread)
```

Comment:
It would be clearer is `execKilled` was inlined here.

```
killResponse.flatMap { success => if (success) list else Nil }(ThreadUtils.sameThread)
```